src/config_generators.js

```diff
@@ -1,11 +1,17 @@
+const yaml = require("js-yaml")
+
 const { getExternalLinks } = require("./helpers")
 
 const generateExternalLinksMenu = courseData => {
-  return `${getExternalLinks(courseData).map((externalLink, index) => {
-    return `[[leftnav]]\n\tname = "${externalLink["title"]}"\n\turl = "${
-      externalLink["url"]
-    }"\n\tweight = ${index * 10 + 1000}`
-  })}`
+  return yaml.safeDump({
+    leftnav: getExternalLinks(courseData).map((externalLink, index) => {
+      return {
+        name:   externalLink["title"],
+        url:    externalLink["url"],
+        weight: index * 10 + 1000
+      }
+    })
```

Comment:
just a style thing, but if you want you can use the single-line style arrow-functions here, both something like




```js
const generateExternalLinksMenu = courseData =>
  yaml.safeDump({
    leftnav: getExternalLinks(courseData).map((externalLink, index) => ({
        name:   externalLink["title"],
        url:    externalLink["url"],
        weight: index * 10 + 1000
    }))
  })
```

Reply:
Good call, I always start out with the curly braces in case I need more than one line and then forget to come back and refactor